File: benchmark/fortran/task4_list.f90
subroutine double_evens(nums, n, result, m)
    integer, intent(in) :: nums(n), n
    integer, intent(out) :: result(n), m
    integer :: i
    m = 0
    do i = 1, n
        if (mod(nums(i), 2) == 0) then
            m = m + 1
            result(m) = nums(i) * 2
        end if
    end do
end subroutine double_evens

File: benchmark/c/task4_list.c
#include <stdlib.h>

int* double_evens(int* nums, int len, int* out_len) {
    int count = 0;
    for (int i = 0; i < len; i++)
        if (nums[i] % 2 == 0) count++;

    int* result = malloc(count * sizeof(int));
    int j = 0;
    for (int i = 0; i < len; i++)
        if (nums[i] % 2 == 0)
            result[j++] = nums[i] * 2;

    *out_len = count;
    return result;
}
